core/src/ledger_cleanup_service.rs

```diff
@@ -270,13 +288,26 @@ mod tests {
         solana_ledger::{blockstore::make_many_slot_entries, get_tmp_ledger_path_auto_delete},
     };
 
+
+    fn flush_blockstore_contents_to_disk(blockstore: Blockstore) -> Blockstore {
```

Comment:
Here is the new helper I made - let's continue the conversation about flush mechanism not working here since we ended up talking about several points in that other thread

Reply:
> Hmmm, calling both [flush_opt()](https://docs.rs/rocksdb/latest/rocksdb/struct.DBCommon.html#method.flush_opt) and [flush_wal()](https://docs.rs/rocksdb/latest/rocksdb/struct.DBCommon.html#method.flush_wal) didn't help; I tried both orders and saw in the LOG file that the manual flush was getting initiated. However, no SST files were created.

I remember some APIs only schedule flushes, but you also need to wait until the flush finishes.

Reply:
> You need to wait until the flush finishes.

I did explicitly set the following to true:
https://docs.rs/rocksdb/latest/rocksdb/struct.FlushOptions.html#method.set_wait

Likewise for the WAL, I set `sync == true` which is supposed to sync it to disk (as opposed to OS buffer I believe)